editor/src/clj/editor/graphics.clj

```diff
@@ -1137,18 +1147,23 @@
                                           (repeat vertex-count matrix-values)))))
 
                   ;; Mesh data doesn't exist. Use the attribute data from the
-                  ;; material or overrides.
-                  (put-renderables! attribute-byte-offset put-attribute-bytes!
-                                    (fn renderable-data->attribute-bytes [renderable-data]
-                                      (let [vertex-count (count (:position-data renderable-data))
-                                            attribute-bytes (get (:vertex-attribute-bytes renderable-data) name-key)
-                                            attribute-byte-count-max (* element-count (buffers/type-size buffer-data-type))]
-                                        ;; Clamp the buffer if the container format is smaller than specified in the material
-                                        (if (> (count attribute-bytes) attribute-byte-count-max)
-                                          (let [attribute-bytes-clamped (byte-array attribute-byte-count-max)]
-                                            (System/arraycopy attribute-bytes 0 attribute-bytes-clamped 0 attribute-byte-count-max)
-                                            (repeat vertex-count attribute-bytes-clamped))
-                                          (repeat vertex-count attribute-bytes))))))
+                  ;; material or overrides. If the material does not declare an
+                  ;; attribute bound by the shader, use a default that makes
+                  ;; sense for the attribute.
+                  (let [attribute-byte-count-max (* element-count (buffers/type-size buffer-data-type))
+                        attribute-data-type (buffer-data-type->attribute-data-type buffer-data-type)
+                        default-attribute-bytes (default-attribute-bytes semantic-type attribute-data-type vector-type normalize)]
+                    (put-renderables!
+                      attribute-byte-offset put-attribute-bytes!
+                      (fn renderable-data->attribute-bytes [renderable-data]
+                        (let [vertex-count (count (:position-data renderable-data))
+                              attribute-bytes (get (:vertex-attribute-bytes renderable-data) name-key default-attribute-bytes)]
+                          ;; Clamp the buffer if the container format is smaller than specified in the material
+                          (if (> (count attribute-bytes) attribute-byte-count-max)
+                            (let [attribute-bytes-clamped (byte-array attribute-byte-count-max)]
+                              (System/arraycopy attribute-bytes 0 attribute-bytes-clamped 0 attribute-byte-count-max)
+                              (repeat vertex-count attribute-bytes-clamped))
+                            (repeat vertex-count attribute-bytes)))))))
```

Comment:
Mostly an indentation change from introducing the surrounding `let`, but we now use the `default-attribute-bytes` if we cannot find any attribute bytes in the `renderable-data`.